CMakeLists.txt

```diff
@@ -22,6 +22,7 @@ if ( ${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" )
         message( FATAL_ERROR "Clang version " ${CMAKE_CXX_COMPILER_VERSION} " not supported. Upgrade to 3.9 or above.")
     else()
         message( STATUS "C++ Compiler is Clang" )
+        add_compile_options(-Wall -Wextra -Wno-deprecated -Werror -pedantic -pedantic-errors)
```

Comment:
We should not have -Wno-deprecated that would allow using deprecated language features.